File: Medium/2125. Number of Laser Beams in a Bank/Solution.cs
using System;

public class Solution {
    public int NumberOfBeams(string[] bank) {
        int totalBeams = 0;
        int prevDeviceCount = 0; // Stores the device count of the last row that had devices

        foreach (string row in bank) {
            // Count devices in the current row
            int currentDeviceCount = row.Length - row.Replace("1", "").Length;

            // If this row has devices...
            if (currentDeviceCount > 0) {
                // Add beams connecting this row to the previous device-row
                totalBeams += (prevDeviceCount * currentDeviceCount);
                // Update the 'previous' count for the next iteration
                prevDeviceCount = currentDeviceCount;
            }
        }

        return totalBeams;
    }
}

File: Medium/2125. Number of Laser Beams in a Bank/Solution.py
class Solution:
    def numberOfBeams(self, bank):
        total_beams = 0
        prev_device_count = 0  # Stores the device count of the last row that had devices

        for row in bank:
            # Count devices in the current row
            current_device_count = row.count('1')
            
            # If this row has devices...
            if current_device_count > 0:
                # Add beams connecting this row to the previous device-row
                total_beams += (prev_device_count * current_device_count)
                # Update the 'previous' count for the next iteration
                prev_device_count = current_device_count
        
        return total_beams